server/src/internalClusterTest/java/org/opensearch/blocks/CreateIndexBlockIT.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.blocks;
+
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+import static org.opensearch.test.OpenSearchIntegTestCase.client;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertBlocked;
+
+public class CreateIndexBlockIT extends OpenSearchIntegTestCase {
+
+    public void testBlockCreateIndex() {
+        try {
+            setCreateIndexBlock("true");
+            assertBlocked(client().admin().indices().prepareCreate("diskguardrails1"), Metadata.CLUSTER_CREATE_INDEX_BLOCK);
+            setCreateIndexBlock("false");
+            assertAcked(client().admin().indices().prepareCreate("diskguardrails2").execute().actionGet());
+        } finally {
```

Comment:
Why this is implemented as part of finally? Can we do it as part of cleanup like after method?

Reply:
Addressed the comments.